lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -738,6 +738,23 @@ def generate_private_key(
     return PrivateKey.from_string(key_bytes.decode())
 
 
+def calculate_relative_datetime(target_time: datetime, fraction: float) -> datetime:
+    """Calculate a datetime that is a given percentage from now to a target time.
+
+    Args:
+        target_time (datetime): The future datetime to interpolate towards.
+        fraction (float): Fraction of the interval from now to target_time (0.0-1.0).
```

Comment:
I think we should not accept anything below `0.5`, as otherwise we do not use up even half of the available life time of the certificate.